client/src/modules/stock/exit/exit.js

```diff
@@ -427,6 +429,53 @@ function StockExitController(
     };
   }
 
+  function buildDescription(entityUuid, invoiceUuid) {
+    const glb = {};
+    return PatientService.read(null, { uuid : entityUuid })
+      .then(([patient]) => {
+        if (patient) {
+          glb.patientDescription = patient.display_name.concat(` (${patient.reference})`);
+        }
+        return invoiceUuid ? PatientInvoiceService.read(null, { uuid : invoiceUuid }) : [];
+      })
+      .then(([invoice]) => {
+        if (invoice) {
+          glb.invoiceDescription = invoice.reference;
+        }
+        return ServiceService.read(null, { uuid : entityUuid });
+      })
+      .then(([service]) => {
+        let description = '';
+        if (service) {
+          glb.serviceDescription = service.name;
+        }
+
+        if (vm.depot.text && glb.patientDescription) {
+          description = $translate.instant('STOCK.EXIT_PATIENT_ADVANCED', {
+            patient : glb.patientDescription,
+            depot : vm.depot.text,
+          });
+        }
+
+        if (vm.depot.text && glb.patientDescription && glb.invoiceDescription) {
+          description = $translate.instant('STOCK.EXIT_PATIENT_ADVANCED_WITH_INVOICE', {
```

Comment:
It looks like your translation key object doesn't change all that much.  Let's put all the keys in a single object and use them for each translation.  Like this:
```js
const i18nKeys = {
  patient : glb.patientDescription,
  depot : vm.depot.text,
  invoice : glb.invoiceDescription
};

// later..
if (x && y) {
  description = $translate.instant('STOCK.EXIT_PATIENT_ADVANCED', i18nKeys);
}

if (y && z) {
 description = $translate.instant('STOCK.EXIT_PATIENT_ADVANCED_WITH_INVOICE', i18nKeys);
}
```